src/analysis/special_functions/trigonometric.lean

```diff
@@ -1754,6 +1755,24 @@ lemma continuous_tan : continuous (λ x : {x | cos x ≠ 0}, tan x) :=
 lemma continuous_on_tan : continuous_on tan {x | cos x ≠ 0} :=
 by { rw continuous_on_iff_continuous_restrict, convert continuous_tan }
 
+lemma range_cos_infinite : (set.range complex.cos).infinite :=
+begin
+  have : (set.range real.cos).infinite,
+  { rw real.range_cos,
+    apply Icc.infinite,
+    norm_num },
+  rw ← set.infinite_coe_iff at this ⊢,
+  delta real.cos at this,
+  apply @infinite.of_injective _ _ this,
+  swap,
+  { rintro ⟨x, hx⟩,
+    refine ⟨x, _⟩,
+    rcases hx with ⟨θ, rfl⟩,
+    simp only [complex.of_real_cos_of_real_re, set.mem_range_self] },
+  { rintro ⟨x, hx⟩ ⟨y, hy⟩ h,
+    simpa only [complex.of_real_inj, subtype.mk_eq_mk] using h }
+end
+
```

Comment:
```suggestion
lemma range_cos_infinite : (set.range complex.cos).infinite :=
begin
  have : (set.range real.cos).infinite,
  { rw real.range_cos,
    apply Icc.infinite,
    norm_num },
  delta real.cos at this,
  rw [range_comp, range_comp] at this,
  exact set.infinite_mono
    (image_subset_range _ (range (coe : ℝ → ℂ))) (λ h, this (set.finite.image _ h)),
end
```
Having to use injectivity felt weird, so I wanted to try without doing it. Maybe that means we should have a lemma saying `(f '' s).infinite -> s.infinite` so that there's no need to go back to `set.finite.image`

Reply:
Thanks!